integration-tests/src/nft_marketplace.rs

```diff
@@ -135,3 +137,105 @@ fn create_class_should_fail_when_relay_chain_location_not_registered() {
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_mint(&ClassType::Auction));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_mint(&ClassType::HydraHeads));
+}
+
+#[test]
+fn nft_permissions_should_allow_marketplace_and_liquidity_mining_class_transfer() {
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&ClassType::Marketplace));
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&ClassType::LiquidityMining));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&ClassType::Redeemable));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&ClassType::Auction));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&ClassType::HydraHeads));
+}
+
+#[test]
+fn nft_permissions_should_allow_marketplace_class_burning() {
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&ClassType::Marketplace));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&ClassType::LiquidityMining));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&ClassType::Redeemable));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&ClassType::Auction));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&ClassType::HydraHeads));
+}
+
+#[test]
+fn nft_permissions_should_allow_marketplace_class_destroying() {
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&ClassType::Marketplace));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&ClassType::LiquidityMining));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&ClassType::Redeemable));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&ClassType::Auction));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&ClassType::HydraHeads));
+}
+
+#[test]
+fn nft_permissions_should_require_deposit_for_marketplace_class_when_created() {
+	assert!(<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&ClassType::Marketplace));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&ClassType::LiquidityMining));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&ClassType::Redeemable));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&ClassType::Auction));
+	assert!(!<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&ClassType::HydraHeads));
+}
+
+#[test]
+fn create_class_should_be_without_deposit_when_created_using_trait() {
```

Comment:
Maybe we could try to leverage the [test-case](https://crates.io/crates/test-case) crate here, so we could remove some test code duplication.

So the test would look like this as a result:
```
#[test_case(ClassType::Marketplace)]
#[test_case(ClassType::LiquidityMining)]
#[test_case(ClassType::Redeemable)]
#[test_case(ClassType::Auction)]
#[test_case(ClassType::HydraHeads)]
fn create_class_should_be_without_deposit_when_created_using_trait(class_type: ClassType) {
	// Arrange
	TestNet::reset();
	Basilisk::execute_with(|| {
		let initial_balance = Balances::free_balance(&AccountId::from(ALICE));

		// Act
		assert_ok!(
			<NFT as CreateTypedClass<AccountId, ClassId, ClassType>>::create_typed_class(
				AccountId::from(ALICE),
				0,
				class_type
			)
		);

                // Assert
		assert_eq!(initial_balance, Balances::free_balance(&AccountId::from(ALICE)));
	});
}
```

How do you like it?

Reply:
Thanks. I like it. I investigated how to iterate over enum but unfortunately it's not that easy in rust (without modifying the enum and introducing new dependency).